### PR TITLE
Add support for codeql

### PR DESCRIPTION
**Pull request by gcabiddu**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
```
Comment (gcabiddu) [inline .github/workflows/codeql.yml:28]: Remove these comments

**Comment by gcabiddu** [comment]
Add badge to readme
https://github.com/intel/qatlib/actions/workflows/codeql.yml/badge.svg

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      run: ACLOCAL_PATH=/usr/share/aclocal ./autogen.sh
+
+    - name: configure
+      run: ./configure
+
+    - name: make
+      run: make -j
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
```
Comment (gcabiddu) [inline .github/workflows/codeql.yml:54]: pin dependency

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
```
[the remark this answers is not in the document]
Comment (gcabiddu) [reply .github/workflows/codeql.yml:8]: You are right.
I started this file from a template from github.
Let me try a test to see if I can safely remove it.

[the remark this answers is not in the document]
Comment (gcabiddu) [reply .github/workflows/codeql.yml]: Fixed

**Review by fionatrahe — APPROVED** [review]

**Pull request merged**